ewfish.py

```diff
@@ -986,3 +986,85 @@ def kill_dead_offers(id_server):
 		id_server,
 		time_now - ewcfg.fish_offer_timeout,
 	))
+
+async def embiggen(cmd):
+	user_data = EwUser(member = cmd.message.author)
+	market_data = EwMarket(id_server = user_data.id_server)
+	item_search = ewutils.flattenTokenListToString(cmd.tokens[1:])
+	item_sought = ewitem.find_item(item_search = item_search, id_user = cmd.message.author.id, id_server = cmd.message.server.id if cmd.message.server is not None else None)
+
+	if cmd.message.channel.name != ewcfg.channel_slimeoidlab:
+		response = "How are you going to embiggen your fish on the side of the street? You’ve got to see a professional for this, man. Head to the SlimeCorp Laboratory, they’ve got dozens of modern day magic potions ‘n shit over there."
+
+	elif item_sought:
+		name = item_sought.get('name')
+		fish = EwItem(id_item = item_sought.get('id_item'))
+		acquisition = fish.item_props.get('acquisition')
+
+		if acquisition != ewcfg.acquisition_fishing:
+			response = "You can only embiggen fishes, dummy. Otherwise everyone would be walking around with colossal nunchucks and huge chicken buckets. Actually, that gives me an idea..."
+
+		else:
+			size = fish.item_props.get('size')
+
+			poudrin_cost = 0
+
+			if size == ewcfg.fish_size_miniscule:
+				poudrin_cost = 2
+
+			if size == ewcfg.fish_size_small:
+				poudrin_cost = 4
+
+			if size == ewcfg.fish_size_average:
+				poudrin_cost = 8
+
+			if size == ewcfg.fish_size_big:
+				poudrin_cost = 16
+
+			if size == ewcfg.fish_size_huge:
+				poudrin_cost = 32
+
+			poudrins_owned = ewitem.find_item_all(item_search = "slimepoudrin", id_user = user_data.id_user, id_server = user_data.id_server)
+			poudrin_amount = len(poudrins_owned)
+
+			if poudrin_cost == 0:
+				response = "Your {} is already as colossal as a fish can get!".format(name)
+
+			elif poudrin_amount < poudrin_cost:
+				response = "You need {} poudrins to embiggen your {}, but you only have {}!!".format(poudrin_cost, name, poudrin_amount)
+
+			else:
+				if size == ewcfg.fish_size_miniscule:
+					fish.item_props['size'] = 'small'
+
+				if size == ewcfg.fish_size_small:
+					fish.item_props['size'] = 'average'
+
+				if size == ewcfg.fish_size_average:
+					fish.item_props['size'] = 'big'
+
+				if size == ewcfg.fish_size_big:
+					fish.item_props['size'] = 'huge'
+
+				if size == ewcfg.fish_size_huge:
+					fish.item_props['size'] = 'colossal'
+
+				fish.persist()
+
+				for delete in range(poudrin_cost):
+					ewitem.item_delete(id_item = "slimepoudrin")
```

Comment:
this won't work. id_item needs to be the item's unique integer id

Reply:
I recommend using the list.pop() method to retrieve and remove poudrins from poudrins_owned